pkg/api/state_test.go

```diff
@@ -0,0 +1,225 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"log"
+	"strconv"
+	"testing"
+)
+
+func TestExample(t *testing.T) {
+	b, err := json.Marshal(map[string]*State{
+		"foo": {
+			Role: "goalkeeper",
+		},
+		"bar": {
+			Role: "goalsaver",
+		},
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println(string(b))
+}
+
+func TestGeneral(t *testing.T) {
+	activeCase := getNewValidState()
```

Comment:
The test should look like this:
```go
type TestCase struct{
   Command Validator
   ShouldError bool
}
for i, tc := range []TestCase{
// Add testcases...
} {
   t.Run(strconv.Itoa(i), func(t *testing.T) {
      err := tc.Command.Validate()
      if tc.ShouldError {
         assert.NotNil(t, err)
      else {
         assert.Nil(t, err)
      }
  })
}
```

you should not define any more functions in `state_test.go` apart from `TestValidate`